ui/apps/ui/src/app/components/filters/filters.service.ts

```diff
@@ -8,26 +8,77 @@ import {
   ITermsFacetResponse,
 } from '@collections/repositories/types';
 import { SearchMetadataRepository } from '@collections/repositories/search-metadata.repository';
-import { Observable, map } from 'rxjs';
+import { Observable, mergeMap } from 'rxjs';
 import { facetToFlatNodes } from '@components/filters/utils';
+import {
+  removeFilterValue,
+  serializeAll,
+} from '@collections/filters-serializers/filters-serializers.utils';
+import { FiltersConfigsRepository } from '@collections/repositories/filters-configs.repository';
+import { ActivatedRoute, Router } from '@angular/router';
+import { IFqMap } from '@collections/services/custom-route.type';
+
+type OriginalStructure = {
+  [field: string]: number | ITermsFacetResponse;
+};
+
+type ConvertedStructure = {
+  [key: string]: string[];
+};
 
 @Injectable({ providedIn: 'root' })
 export class FilterService {
   constructor(
     private _fetchDataService: FetchDataService,
-    private _searchMetadataRepository: SearchMetadataRepository
+    private _searchMetadataRepository: SearchMetadataRepository,
+    private _filtersConfigsRepository: FiltersConfigsRepository,
+    private _route: ActivatedRoute,
+    private _router: Router
   ) {}
 
-  fetchTreeNodes$(
+  convertFacets(facets: OriginalStructure): ConvertedStructure {
+    const result: ConvertedStructure = {};
+
+    for (const key in facets) {
+      const facet = facets[key];
+      if (typeof facet !== 'number' && 'buckets' in facet) {
+        // Zakładamy, że facet.buckets.map zwraca (string | number)[]
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        result[key] = facet.buckets.map((bucket: { val: any }) => bucket.val);
+      }
+    }
+
+    return result;
+  }
+
+  /*fetchTreeNodes$(
```

Comment:
Seems like a dev stuff

Reply:
yup, but i preffer to keep it there at least for a sec